pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -298,24 +298,32 @@ public boolean isSuperTypeOf(ColumnDataType subTypeCandidate) {
     public DataType toDataType() {
       switch (this) {
         case INT:
+        case INT_ARRAY:
```

Comment:
why this change? let's dont mix array support in here if we can make that separate

Reply:
The issue is that json_extract_key returns array.

Reply:
i see. not sure if there's going to be any side effect of this change going forward with ARRAY supports. 
can you add support like
`SELECT cardinality(JSONEXTRACTKEY(...))` and see if that works?

Reply:
Added a sample query in unit test.